pandas/core/indexes/multi.py

```diff
@@ -4135,6 +4135,33 @@ def isin(self, values, level=None) -> npt.NDArray[np.bool_]:
     # base class "Index" defined the type as "Callable[[Index, Any, bool], Any]")
     rename = Index.set_names  # type: ignore[assignment]
 
+    def difference(self, other, sort=None):
```

Comment:
this seems really unlikely to be the right place to handle this.  if you step through the existing implementation, where is the first step that goes wrong?